tests/test_put_file.py

```diff
@@ -56,3 +60,61 @@ def test_no_change_postcommit(
     )
     assert commits.results[0] == latest_commit
     # in particular, this test asserts that no API exception happens in postcommit.
+
+def test_implicit_branch_creation(
+    random_file_factory: RandomFileFactory,
+    fs: LakeFSFileSystem,
+    repository: str,
+    temp_branch: str):
+
+    # Create Changes to commit. 
+    random_file = random_file_factory.make()
+    lpath = str(random_file)
+
+
+    with fs.scope(create_branch_ok = True):
+        #Pushing to an existing branch
+        rpath = f"{repository}/{temp_branch}/{random_file.name}"
+        fs.put(lpath, rpath)
+        commits = fs.client.commits_api.log_branch_commits(
+            repository=repository,
+            branch=temp_branch,
+        )
+        latest_commit = commits.results[0]  # commit log is ordered branch-tip-first
+        assert latest_commit.message == f"Add file {random_file.name}"
```

Comment:
This test is redundant with the test above `test_put_with_default_commit_hook` as `create_branch_ok` defaults to `True`. It is not explicitly set above, however in the above test for the commit hook whereas here it is. 

The same is true for the test of pushing to an existing branch with `create_branch_ok = False` as then nothing unexpected should happen.  

Should I keep or remove these two tests (creates commit in existing branch with `create_branch_ok = True`  and creates commit in existing branch with `create_branch_ok = False`)?

Reply:
Remove please. Anything on existing branches is irrelevant for the scope of `create_branch_ok`.